Shared/Realm.Shared/Realm.cs

```diff
@@ -985,5 +973,80 @@ public void Close()
         }
 
         #endregion
+
+        private class RealmState
+        {
+            #region static
+
+            #if __IOS__
+            [MonoPInvokeCallback(typeof(NativeCommon.NotifyRealmCallback))]
+            #endif
+            public static void NotifyRealmChanged(IntPtr stateHandle)
+            {
+                var gch = GCHandle.FromIntPtr(stateHandle);
+                ((RealmState)gch.Target).NotifyChanged(EventArgs.Empty);
```

Comment:
This is correct - the handle is only freed once the last realm it caches is closed.